core/src/test/java/apoc/map/MapsTest.java

```diff
@@ -317,6 +317,36 @@ public void testFlattenWithDelimiter() {
         });
     }
 
+    @Test
+    public void testUnflattenRoundrip() {
+        List<Map<String, Object>> innerNestedListMap = List.of(map("somekey", "someValue", "somenumeric", 123), 
+                map("keyFoo", "valueFoo"),
+                map("keyBar", "valueBar"));
+        Map<String, Object> nestedMap = map("anotherkey", "anotherValue", "nested", innerNestedListMap);
+        final Map<String, Object> expectedMap = map("string", "value", "int", 10, "nested", nestedMap);
+
+        TestUtil.testCall(db, "WITH apoc.map.flatten($expectedMap) AS flattedMap RETURN apoc.map.unflatten(flattedMap) AS value",
+                map("expectedMap", expectedMap),
+                r -> assertEquals(expectedMap, r.get("value")));
+    }
+
+    @Test
+    public void testUnflattenRoundripWithCustomDelimeter() {
```

Comment:
```suggestion
    public void testUnflattenRoundtripWithCustomDelimiter() {
```

Reply:
Nit: maybe do not test the round trip conversion here but start with an already-flattened map and call only unflatten with the custom delimiter?